### PR TITLE
[Inference] Benchmarking rotary embedding and add a fetch function

### PR DESCRIPTION
**Pull request by CjhHa1**
## 📌 Checklist before creating the PR

- [ ] I have created an issue for this PR for traceability
- [x] The title follows the standard format: `[doc/gemini/tensor/...]: A concise description`
- [x] I have added relevant tags if possible for us to better distinguish different PRs

## 🚨 Issue number

> Link this PR to your issue with words like fixed to automatically close the linked issue upon merge
>
> e.g. `fixed #1234`, `closed #1234`, `resolved #1234`

## 📝 What does this PR do?

> Summarize your work here.
> if you have any plots/diagrams/screenshots/tables, please attach them here.

## 💥 Checklist before requesting a review

- [ ] I have linked my PR to an issue ([instruction](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue))
- [ ] My issue clearly describes the problem/feature/proposal, with diagrams/charts/table/code if possible
- [ ] I have performed a self-review of my code
- [x] I have added thorough tests.
- [x] I have added docstrings for all the functions/methods I implemented

## ⭐️ Do you enjoy contributing to Colossal-AI?

- [x] 🌝 Yes, I do.
- [ ] 🌚 No, I don't.

Tell us more if you don't enjoy contributing to Colossal-AI.

Benchmarking:
image

### CODE REVIEW

File: colossalai/inference/modeling/models/llama.py [hunk 1]
```diff
@@ -39,6 +39,16 @@ def apply_rotary_pos_emb(q, k, cos, sin, position_ids):
     return q_embed, k_embed
 
 
+def get_cos_sin(lengths, cos_cache, sin_cache):
```
Comment (yuehuayingxueluo) [inline colossalai/inference/modeling/models/llama.py:42]: How can we use this cache in llama model？

Comment (CjhHa1) [reply colossalai/inference/modeling/models/llama.py:42]: I will add docsrting

Comment (CjhHa1) [reply colossalai/inference/modeling/models/llama.py:42]: done

**Review by yuehuayingxueluo — APPROVED** [review]

**Comment by CjhHa1** [comment]
note: `get_cos_sin` function is implemented in another PR and will be replaced by triton kernel

**Pull request merged**